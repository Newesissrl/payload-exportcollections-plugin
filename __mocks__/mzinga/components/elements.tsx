import React from "react";
//this below to avoid vscode auto-organize import to remove 'React' dependency
type _react = typeof React;
export const Button = (args: any) => {
  return <button {...args}></button>;
};
