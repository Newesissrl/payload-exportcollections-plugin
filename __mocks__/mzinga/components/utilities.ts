export const useConfig = () => {
  return {
    serverURL: "http://localhost:3033",
    routes: {
      api: "/api",
    },
  };
};
