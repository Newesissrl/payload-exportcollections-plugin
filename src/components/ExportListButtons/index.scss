.buttons {
  display: flex;
  justify-content: flex-end;
  & button {
    margin: 0 8px;
    &:last-of-type {
      margin-right: 0;
    }
  }
  .hidden {
    display: none;
  }
}
