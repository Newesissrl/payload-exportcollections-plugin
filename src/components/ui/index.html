<!DOCTYPE html>
<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, user-scalable=no"
    />
    <link rel="icon" href="data:," data-placeholder-favicon />
  </head>

  <body>
    <div id="app"></div>
  </body>
</html>
