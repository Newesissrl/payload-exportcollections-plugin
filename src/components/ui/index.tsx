// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore - need to do this because this file doesn't actually exist
import React from "react";
import { createRoot } from "react-dom/client";

const Root = () => {
  return <></>;
};

const container = document.getElementById("app");
const root = createRoot(container); // createRoot(container!) if you use TypeScript
root.render(<Root />);

// Needed for Hot Module Replacement
if (
  typeof module !== "undefined" &&
  module &&
  "hot" in module &&
  typeof module.hot === "object" &&
  "accept" in module.hot &&
  typeof module.hot.accept === "function"
) {
  module.hot.accept();
}
