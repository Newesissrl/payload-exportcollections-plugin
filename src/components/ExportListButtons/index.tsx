import { Button } from "mzinga/components/elements";
import { useConfig } from "mzinga/components/utilities";
import { useSearchParams } from "mzinga/dist/admin/components/utilities/SearchParams";
import qs from "qs";
import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles({
  buttons: {
    display: "flex",
    justifyContent: "flex-end",
    "& button": {
      margin: "0 8px",
      "&:last-of-type": {
        marginRight: 0,
      },
    },
    "& .hidden": {
      display: "none",
    },
  },
});
const ExportTypes = {
  JSON: "json",
  CSV: "csv",
};
const DownloadExportFile = ({ slug, type, fileName }) => {
  const {
    routes: { api },
  } = useConfig();

  return (
    <a
      className="hidden"
      target="_download"
      id={`download-${type}`}
      href={`${api}/${slug}/download-exports?filePath=${fileName}`}
    >
      {`download-${type}`}
    </a>
  );
};
const ExportButton = ({ onClick, type, disabled }) => {
  const { t } = useTranslation("general");
  return (
    <Button
      type="submit"
      onClick={() => onClick(type)}
      disabled={disabled}
      data-testid={`export-list-${type}`}
    >
      {t(`export-list-${type}`)}
    </Button>
  );
};
export const ExportListButtons = ({ collection }) => {
  const params = useSearchParams();
  const classes = useStyles();
  const [fileNameCSV, setFileNameCSV] = useState(null);
  const [fileNameJSON, setFileNameJSON] = useState(null);
  const [userInfo, setUserInfo] = useState(null);
  const [isExporting, setIsExporting] = useState(false);
  const { slug } = collection;
  const config = useConfig();
  if (!(config && config.serverURL)) {
    return null;
  }
  const {
    routes: { api },
    serverURL,
  } = config;
  const save = async (exportType) => {
    setIsExporting(true);
    const query = qs.stringify({
      ...params,
      id: userInfo.id,
      type: exportType,
      slug,
    });
    const filePath = await fetch(
      `${serverURL}${api}/${slug}/save-exports?${query}`,
      {
        method: "GET",
      }
    ).then((res) => res.text());
    setIsExporting(false);
    if (exportType === ExportTypes.CSV) {
      setFileNameCSV(filePath);
      return;
    }
    setFileNameJSON(filePath);
  };
  useEffect(() => {
    if (!fileNameCSV) {
      return;
    }
    document.getElementById("download-csv").click();
  }, [fileNameCSV]);
  useEffect(() => {
    if (!fileNameJSON) {
      return;
    }
    document.getElementById("download-json").click();
  }, [fileNameJSON]);
  useEffect(() => {
    const getUserInfo = async () => {
      const userData = await fetch(`${serverURL}${api}/users/me`).then((res) =>
        res.json()
      );
      setUserInfo(userData.user);
    };
    getUserInfo();
  }, [serverURL, api, setUserInfo]);
  return (
    userInfo && (
      <div className={classes.buttons}>
        {fileNameCSV && (
          <DownloadExportFile fileName={fileNameCSV} slug={slug} type={"csv"} />
        )}
        <ExportButton
          type={ExportTypes.CSV}
          onClick={save}
          disabled={isExporting}
        />
        {fileNameJSON && (
          <DownloadExportFile
            fileName={fileNameJSON}
            slug={slug}
            type={"json"}
          />
        )}
        <ExportButton
          type={ExportTypes.JSON}
          onClick={save}
          disabled={isExporting}
        />
      </div>
    )
  );
};
