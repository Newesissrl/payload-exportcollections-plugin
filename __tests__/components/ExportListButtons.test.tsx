import React from "react";
import renderer from "react-test-renderer";
import { ExportListButtons } from "../../src/components/ExportListButtons";

describe("ExportListButtons", () => {
  it("should render", () => {
    const component = renderer.create(<ExportListButtons collection="users" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
