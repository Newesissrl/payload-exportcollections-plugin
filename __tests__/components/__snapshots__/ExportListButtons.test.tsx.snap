// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportListButtons should render 1`] = `null`;
